### PR TITLE
simple-linked-list: Re-implementation

### PR DESCRIPTION
**Pull request by Insti**
Ref: [Simple linked list](https://github.com/exercism/x-common/blob/master/exercises/simple-linked-list/description.md)

The previous implementation exposed way too many implementation details
and methods such as `peek` that were not even mentioned in the Readme.

This implementation is much simpler and only tests the methods
mentioned in the Readme\*.

Element.new(value)
Element.datum
Element.next
Element.next=

SimpleLinkedList.new(array=[])
SimpleLinkedList.to_a
SimpleLinkedList.push
SimpleLinkedList.pop # (Last In, First Out)
SimpleLinkedList.reverse!

\* push and pop are not explicitly mentioned in the readme, but it's
hard to operate a LIFO stack without them.

I have also added the BookKeeping information.

### CODE REVIEW

**Comment by bmulvihill** [comment]
Yeah I was reading this and was kind of confused, are we implementing a Stack or a Singly Linked List? A stack can be implemented using a linked list, but you could also implement a stack with an Array.

File: exercises/simple-linked-list/.version [hunk 1]
```diff
@@ -0,0 +1 @@
+2
```
Comment (kotp) [inline exercises/simple-linked-list/.version]: Why is this all of a sudden version 2 from version 0?

File: exercises/simple-linked-list/example.rb [hunk 2]
```diff
@@ -1,68 +1,59 @@
...
 
-  def self.from_a(a)
-    new.tap do |list|
-      a.to_a.reverse_each do |item|
-        list.push(item)
-      end
-    end
-  end
+module BookKeeping
+  VERSION = 2
```
Comment (kotp) [inline exercises/simple-linked-list/example.rb]: Should be version 1, from non-tracked version.

File: exercises/simple-linked-list/simple_linked_list_test.rb [hunk 3]
```diff
@@ -1,189 +1,165 @@
...
+  #
+  # If you are curious, read more about constants on RubyDoc:
+  # http://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/constants.html
 
-  def test_roundtrip_10_element_array
+  def test_bookkeeping
     skip
-    data = (1..10).to_a
-    assert_equal data, SimpleLinkedList.from_a(data).to_a
+    assert_equal 2, BookKeeping::VERSION
```
Comment (kotp) [inline exercises/simple-linked-list/simple_linked_list_test.rb]: also here.

**Comment by Insti** [comment]
> Should this be converted to a generator version?

If it should it will be a different PR. At the moment there is no `canonical-data.json`

**Comment by kotp** [comment]
:heavy_check_mark: Good project once this is in.

**Comment by Insti** [comment]
I've merged this in, we'll see how the students react.
I'll keep an eye on the submissions over the next few days.

**Pull request merged**